### PR TITLE
[google_workspace] - Updated google drive event schema and mappings to incorporate missing fields and avoid conflicts

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
- Bug

## Proposed commit message
**WHAT:**  Updated google drive event schema and mappings to incorporate missing fields and avoid conflicts.
**WHY:** It seems that there are some drive events that contain certain fields which were unmapped. The fields in question are as follows: - 
```js
 1. actor_is_collaborator_account
 2. is_encrypted
 3. owner_is_team_drive
 4. target_user
 5. target
``` 
as defined [here](https://developers.google.com/admin-sdk/reports/v1/appendix/activity/drive). The  only exception is the field **owner_is_team_drive**, which seems to be present in certain events but does not have an explicit definition anywhere. This also creates a scenario where for certain [logs](https://developers.google.com/admin-sdk/reports/v1/appendix/activity/drive#email_as_attachment) the **target_user** and **target** are both present in the source event, where they have different meanings and are not the same, hence **target** has to be mapped conditionally on our end.

## Sample Log:
```json
{
    "actor": {
        "email": "[john.doe@example.com](mailto:john.doe@example.com)",
        "profileId": "987654"
    },
    "etag": "-xyz1234567890/abcdefg",
    "events": {
        "name": "email_as_attachment",
        "parameters": [
            {
                "name": "target",
                "value": "[jane.smith@example.org](mailto:jane.smith@example.org)"
            },
            {
                "name": "target_user",
                "value": "[manager@example.com](mailto:manager@example.com)"
            },
            {
                "boolValue": true,
                "name": "primary_event"
            },
            {
                "boolValue": true,
                "name": "billable"
            },
            {
                "boolValue": false,
                "name": "owner_is_shared_drive"
            },
            {
                "name": "owner",
                "value": "[admin@example.co](mailto:admin@example.co)"
            },
            {
                "name": "doc_id",
                "value": "doc123-456"
            },
            {
                "name": "doc_type",
                "value": "spreadsheet"
            },
            {
                "boolValue": false,
                "name": "is_encrypted"
            },
            {
                "name": "doc_title",
                "value": "Quarterly Report"
            },
            {
                "name": "visibility",
                "value": "shared_externally"
            },
            {
                "boolValue": false,
                "name": "actor_is_collaborator_account"
            },
            {
                "boolValue": false,
                "name": "owner_is_team_drive"
            }
        ],
        "type": "access"
    },
    "id": {
        "applicationName": "drive",
        "customerId": "customer12345",
        "time": "2024-07-29T12:34:56.789Z",
        "uniqueQualifier": "4567890"
    },
    "kind": "admin#reports#activity"
}
```
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -182,6 +182,7 @@ processors:
   - rename:
       field: google_workspace.drive.target_user
       target_field: google_workspace.drive.target
+      if: ctx.google_workspace.drive.target == null
```
Comment (kcreddy) [inline packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml:185]: Should this be `ctx.google_workspace?.drive?.target == null`? Or at this point we already know they are not null?

**Review by kcreddy — APPROVED** [review]
Minor clarification. LGTM

File: packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (ShourieG) [reply packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml:185]: @kcreddy, at this point **google_workspace.drive** will always exist as its being created by a script at the upper level.

**Pull request merged**